### PR TITLE
Allow schema markdown docs to be remote loaded

### PR DESCRIPTION
**Pull request by vijaypemmaraju**

### CODE REVIEW

File: source/layouts/layout.erb [hunk 1]
```diff
@@ -16,11 +16,7 @@ under the License.
 <% language_tabs = current_page.data.language_tabs || [] %>
 <% page_content = yield %>
 <%
-  if current_page.data.includes
-    current_page.data.includes.each do |include|
-      page_content += partial("includes/#{include}")
-    end
-  end
+  page_content += partial("includes/_graphql.md.erb")
```
Comment (vijaypemmaraju) [inline source/layouts/layout.erb]: we don't have any other includes right now, so i've just hardcoded the graphql partial. We can change this in the future if we want to pull out other things.

File: config.rb [hunk 2]
```diff
@@ -49,6 +51,19 @@
   # activate :gzip
 end
 
+helpers do
+  def remote_markdown
+    # TODO: @vijaypemmmaraju change this to the URL of the actual docs
+    response = HTTParty.get('https://s3-us-west-2.amazonaws.com/developer.grooveapp.com/production/public-api-documentation/docs.md')
```
Comment (vijaypemmaraju) [inline config.rb]: @jaebradley is there a way to hit this through `developer.grooveapp.com` now?

**Comment by vijaypemmaraju** [comment]
@jaebradley I just realized something. If we update the schema, the remote file will change, sure, but don't we need to also trigger a new build here so that s3 gets the updated docs?

File: source/javascripts/app/_search.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 //= require ../lib/_lunr
 //= require ../lib/_jquery
 //= require ../lib/_jquery.highlight
-;(function () {
+function setupSearch() {
```
Comment (vijaypemmaraju) [inline source/javascripts/app/_search.js:4]: notice this is no longer an iffy

File: source/javascripts/all.js.es6 [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
 
 $(function() {
   loadToc($('#toc'), '.toc-link', '.toc-list-h2', 10);
   setupLanguages($('body').data('languages'));
-  $('.content').imagesLoaded( function() {
+  $('.content').imagesLoaded(async() => {
+    await remoteLoad();
     window.recacheHeights();
     window.refreshToc();
+    setupSearch();
```
Comment (vijaypemmaraju) [inline source/javascripts/all.js.es6:19]: we want to make sure remote docs are loaded _before_ building search index

File: source/javascripts/app/_remote_load.js.es6 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      </li>`
+  ).join('');
+}
+
+async function remoteLoad() {
+  let text = '<p style="color:red">Unable to load API docs</p>';
+
+  try {
+    const response = await fetch('https://s3-us-west-2.amazonaws.com/developer.grooveapp.com/production/public-api-documentation/docs.html')
+    if (response.status >= 400) {
```
Comment (jaebradley) [inline source/javascripts/app/_remote_load.js.es6]: I think a more exacting check may be

```javascript
if (response.status === 200) {
  text = await response.text();
}
```

File: source/javascripts/all.js.es6 (hunk 4 above)
Comment (jaebradley) [reply source/javascripts/all.js.es6:19]: If it's not obvious from the code why this dependent behavior is necessary, maybe it's worth it to add some comments.

**Review by jaebradley — COMMENTED** [review]
**EDIT: The behavior described in this comment was due to a cached version of the remote loaded assets. It looks fine right now / after hard-reloading**

# Difference in behavior

Unfortunately, it looks like there's some pretty significant difference in behavior (font-sizing, etc.) that would be nice to keep.

## `master` right now

Screen Shot 2019-04-29 at 3 57 19 PM

## This branch

Screen Shot 2019-04-29 at 3 57 38 PM

I think the new font sizes are kind've untenable for public documentation (the `h4`s look _really_ small). 

Is there some way that this can be rectified while still remote-loading the published docs?

If not, maybe the best solution is that we need to generate the markdown manually when we update the public API definition.

File: source/javascripts/app/_remote_load.js.es6 (hunk 5 above)
Comment (vijaypemmaraju) [reply source/javascripts/app/_remote_load.js.es6]: If aws someday changed their status code to some 2xx that's not 200, but still included the response as expected, would we want to throw?

**Review by jaebradley — APPROVED** [review]
remote

File: source/javascripts/app/_remote_load.js.es6 (hunk 5 above)
Comment (jaebradley) [reply source/javascripts/app/_remote_load.js.es6]: @vijaypemmaraju it looks like you implemented this but just for completeness sake I'll address the comment!

If AWS ever did this it would a breaking change for their API and I assume we'd get a bit of a heads up.

Regardless, here's why I prefer the explicit `200` check - a `204` would not throw but then  the content would not be the default content, not to mention all the `300`s that would not throw that we'd potentially use the response text of.

Basically, we will only use the response text when it's a `200` response - the only response code we can bank on that should return the text we want. So then, if we know that, I don't see any downside to checking for that "bankable" response code directly.
